File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/Chat_Notification.java
package org.emulinker.kaillera.controller.v086.protocol;

import com.google.auto.value.AutoValue;
import org.emulinker.kaillera.controller.messaging.MessageFormatException;

@AutoValue
public abstract class Chat_Notification extends Chat {
  private static final String DESC = "Chat Notification";

  public static AutoValue_Chat_Notification create(
      int messageNumber, String username, String message) throws MessageFormatException {
    V086Message.validateMessageNumber(messageNumber, DESC);
    return new AutoValue_Chat_Notification(messageNumber, ID, DESC, username, message);
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/Chat_Notification.kt
package org.emulinker.kaillera.controller.v086.protocol

import org.emulinker.kaillera.controller.messaging.MessageFormatException

data class Chat_Notification
    @Throws(MessageFormatException::class)
    constructor(
        override val messageNumber: Int, override val username: String, override val message: String
    ) : Chat() {

  override val description = DESC

  override val messageId = ID

  companion object {
    private const val DESC = "Chat Notification"
  }
}
